File: src/main/scala/com/github/kmizu/yapp/runtime/SimpleCharacterSet.scala
package com.github.kmizu.yapp.runtime

final class SimpleCharacterSet(chars: Char*) extends CharacterSet {
  private[this] val table = new Array[Boolean](Character.MAX_VALUE)

  {
    for (ch <- chars) table(ch) = true
  }

  def contains(ch: Char): Boolean =  table(ch)

  def add(ch: Char): Unit = table(ch) = true

}

File: src/main/java/com/github/kmizu/yapp/runtime/SimpleCharacterSet.java
package com.github.kmizu.yapp.runtime;

public final class SimpleCharacterSet implements CharacterSet {
  private final boolean[] table = new boolean[Character.MAX_VALUE];
  public SimpleCharacterSet(char... chars) {
    for(char ch : chars) table[ch] = true;
  }
  public boolean contains(char ch) {
    return table[ch];
  }
  public void add(char ch) {
    table[ch] = true;
  }
}
